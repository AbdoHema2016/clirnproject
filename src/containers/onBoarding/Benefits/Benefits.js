import {View} from 'react-native';
import React, {Component} from 'react';
import {
  IndicatorViewPager,
  PagerDotIndicator,
} from 'react-native-best-viewpager';
import CBenefit from './component/CBenefit';
import Style from './style';
import CButton from '../../../components/cButton';
import {Constants} from '../../../utilities';
import {translate} from '../../../Localization';

import {logAnalyticsEvent} from '../../../utilities/Firebase';

const {LOCAL_PATH, testIds, analyticsIds} = Constants;

export default class Benefits extends Component {
  constructor(props) {
    super(props);
    this.state = {
      currentPage: 0,
    };
  }

  setPage = () => {
    if (this._setPage._currentIndex === 1) {
      this.props.navigation.navigate('PersonalDetails');
      logAnalyticsEvent(analyticsIds.start_personal_details);
    } else {
      this._setPage._goToNextPage();
      logAnalyticsEvent(analyticsIds.onboard_benefit);
    }
  };

  goToSignIn = () => {
    this.props.navigation.navigate('SignIn');
  };

  render() {
    return (
      <View style={Style.container}>
        <IndicatorViewPager
          style={Style.container}
          indicator={this._renderDotIndicator()}
          ref={(ref) => (this._setPage = ref)}>
          <View style={Style.innerContainers}>
            <CBenefit
              title={translate('BENEFIT_SCREEN_STRINGS.BENEFIT_FIRST_TITLE')}
              subTitle={translate(
                'BENEFIT_SCREEN_STRINGS.BENEFIT_FIRST_SUBTITLE',
              )}
              imagePath={LOCAL_PATH.BENEFIT1_ICON}
            />
          </View>
          <View style={Style.innerContainers}>
            <CBenefit
              title={translate('BENEFIT_SCREEN_STRINGS.BENEFIT_SECOND_TITLE')}
              subTitle={translate(
                'BENEFIT_SCREEN_STRINGS.BENEFIT_SECOND_SUBTITLE',
              )}
              imagePath={LOCAL_PATH.BENEFIT2_ICON}
            />
          </View>
        </IndicatorViewPager>
        <View style={Style.buttonsContainer}>
          <CButton
            testID={testIds.continueToNextBenefit}
            accessibilityLabel={testIds.continueToNextBenefit}
            text={'Continue'}
            textStyle={Style.whiteText}
            onPress={this.setPage}
            buttonContainerStyle={Style.blackButton}
          />
          <CButton
            testID={testIds.gotoSignInFromBenefits}
            accessibilityLabel={testIds.gotoSignInFromBenefits}
            text={'Sign In'}
            textStyle={Style.blackText}
            onPress={this.goToSignIn}
            buttonContainerStyle={Style.whiteButton}
          />
        </View>
      </View>
    );
  }

  _renderDotIndicator() {
    return (
      <PagerDotIndicator
        dotStyle={Style.unSelectedDotStyle}
        selectedDotStyle={Style.selectedDotStyle}
        pageCount={2}
      />
    );
  }
}
