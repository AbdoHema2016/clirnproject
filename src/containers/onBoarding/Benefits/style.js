import {StyleSheet} from 'react-native';
import {Layout} from '../../../utilities';

const {LAYOUT_CONSTRAINTS, HEX_COLOR_CODES} = Layout;

const style = StyleSheet.create({
  container: {
    flex: 1,
  },
  innerContainers: {
    height: LAYOUT_CONSTRAINTS.WINDOW_HEIGHT,
  },
  selectedDotStyle: {
    bottom:
      LAYOUT_CONSTRAINTS.WINDOW_HEIGHT > 700
        ? LAYOUT_CONSTRAINTS.WINDOW_HEIGHT * 0.29
        : 180,
    height: 16,
    width: 16,
    borderRadius: 12,
    backgroundColor: HEX_COLOR_CODES.GREEN_BLUE,
  },
  unSelectedDotStyle: {
    bottom:
      LAYOUT_CONSTRAINTS.WINDOW_HEIGHT > 700
        ? LAYOUT_CONSTRAINTS.WINDOW_HEIGHT * 0.29
        : 180,
    height: 16,
    width: 16,
    borderRadius: 12,
    backgroundColor: HEX_COLOR_CODES.WHITE,
  },
  blackButton: {
    backgroundColor: HEX_COLOR_CODES.BLACK_BUTTON,
  },
  buttonsContainer: {
    alignSelf: 'center',
    position: 'absolute',
    bottom:
      LAYOUT_CONSTRAINTS.WINDOW_HEIGHT > 800
        ? LAYOUT_CONSTRAINTS.WINDOW_HEIGHT * 0.109
        : 32,
    justifyContent: 'space-between',
    width: LAYOUT_CONSTRAINTS.WINDOW_WIDTH - 28,
    height: 116,
  },
  whiteButton: {
    backgroundColor: HEX_COLOR_CODES.WHITE,
    marginTop: 16,
  },
  whiteText: {
    color: HEX_COLOR_CODES.WHITE,
  },
  blackText: {
    color: HEX_COLOR_CODES.BLACK,
  },
});

export default style;
