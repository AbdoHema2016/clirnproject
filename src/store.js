import {createStore, applyMiddleware} from 'redux';
import createSagaMiddleware from 'redux-saga';
import {createLogger} from 'redux-logger';
import rootReducer from './reducers';
import {rootSaga} from './sagas';
const sagaMiddleware = createSagaMiddleware();
const isDevelopmentEnv =
  process && process.env && process.env.NODE_ENV === 'development';
const isProductionEnv =
  process && process.env && process.env.NODE_ENV === 'production';
const logStoreInfo = false;
const middlewares = [sagaMiddleware];
function actionLogger({getState}) {
  return (next) => (action) => {
    let allowed = [
      'jumio',
      'notificationsReducer',
      'otp',
      'companies',
      'signIn',
      'updateContactDetailsReducer',
      'userProfile',
      'vaccinations',
    ];
    let state = getState();
    let loggedStates = Object.keys(state)
      .filter((key) => allowed.includes(key))
      .reduce((obj, key) => {
        return {
          ...obj,
          [key]: state[key],
        };
      }, {});
    console.info({actionType: action.type, loggedStates});
    const returnValue = next(action);
    return returnValue;
  };
}
if (isDevelopmentEnv && logStoreInfo) {
  middlewares.push(createLogger());
}
if (isProductionEnv) {
  middlewares.push(actionLogger);
}

const store = createStore(rootReducer, applyMiddleware(...middlewares));
sagaMiddleware.run(rootSaga);
export default store;
