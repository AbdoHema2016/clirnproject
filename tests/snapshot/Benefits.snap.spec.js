import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import store from '../../src/store';
import Benefits from '../../src/containers/onBoarding/Benefits/Benefits';

jest.mock('@react-navigation/native', () => ({
  withNavigationFocus: (component) => component,
}));

jest.useFakeTimers();

describe('Benefits screen', () => {
  const props = {
    navigation: {
      setOptions: jest.fn(),
    },
  };
  it('Benefits OTP', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Benefits {...props} />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
