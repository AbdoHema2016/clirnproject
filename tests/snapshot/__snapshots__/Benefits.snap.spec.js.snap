// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Benefits screen Benefits OTP 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View>
    <View
      style={
        Object {
          "height": 1334,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "height": "100%",
            "width": 750,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontFamily": "SofiaProRegular",
                "fontSize": 24,
                "marginTop": 360.18,
                "textAlign": "center",
              },
            ]
          }
        >
          [missing "en.BENEFIT_SCREEN_STRINGS.BENEFIT_FIRST_TITLE" translation]
        </Text>
        <Text
          style={
            Array [
              Object {
                "alignSelf": "center",
                "color": "#8D8C91",
                "fontFamily": "SofiaProRegular",
                "fontSize": 18,
                "marginTop": 16,
                "textAlign": "center",
                "width": 288,
              },
            ]
          }
        >
          [missing "en.BENEFIT_SCREEN_STRINGS.BENEFIT_FIRST_SUBTITLE" translation]
        </Text>
        <View
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "alignSelf": "center",
                "height": 120,
                "marginTop": 626.98,
                "position": "absolute",
                "width": 120,
              },
            ]
          }
        >
          <FastImageView
            resizeMode="cover"
            source={
              Object {
                "cache": "immutable",
                "uri": undefined,
              }
            }
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "height": 1334,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "height": "100%",
            "width": 750,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontFamily": "SofiaProRegular",
                "fontSize": 24,
                "marginTop": 360.18,
                "textAlign": "center",
              },
            ]
          }
        >
          [missing "en.BENEFIT_SCREEN_STRINGS.BENEFIT_SECOND_TITLE" translation]
        </Text>
        <Text
          style={
            Array [
              Object {
                "alignSelf": "center",
                "color": "#8D8C91",
                "fontFamily": "SofiaProRegular",
                "fontSize": 18,
                "marginTop": 16,
                "textAlign": "center",
                "width": 288,
              },
            ]
          }
        >
          [missing "en.BENEFIT_SCREEN_STRINGS.BENEFIT_SECOND_SUBTITLE" translation]
        </Text>
        <View
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "alignSelf": "center",
                "height": 120,
                "marginTop": 626.98,
                "position": "absolute",
                "width": 120,
              },
            ]
          }
        >
          <FastImageView
            resizeMode="cover"
            source={
              Object {
                "cache": "immutable",
                "uri": undefined,
              }
            }
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          />
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignSelf": "center",
        "bottom": 145.406,
        "height": 116,
        "justifyContent": "space-between",
        "position": "absolute",
        "width": 722,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderRadius": 4,
            "height": 52,
            "justifyContent": "center",
            "shadowColor": "#DDDDDD",
            "shadowOffset": Object {
              "height": 8,
              "width": 0,
            },
            "shadowOpacity": 0.75,
            "shadowRadius": 4,
          },
          Object {
            "backgroundColor": "#565655",
          },
        ]
      }
    >
      <View
        accessibilityLabel="ContinueToNextBenefit"
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": undefined,
            "borderRadius": 4,
            "height": "100%",
            "justifyContent": "center",
            "opacity": 1,
            "width": 718,
          }
        }
        testID="ContinueToNextBenefit"
      >
        <Text
          numberOfLines={1}
          style={
            Array [
              Object {
                "fontFamily": "SofiaProRegular",
                "fontSize": 17,
                "textAlign": "center",
              },
              Object {
                "color": "#FFFFFF",
              },
            ]
          }
        >
          Continue
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderRadius": 4,
            "height": 52,
            "justifyContent": "center",
            "shadowColor": "#DDDDDD",
            "shadowOffset": Object {
              "height": 8,
              "width": 0,
            },
            "shadowOpacity": 0.75,
            "shadowRadius": 4,
          },
          Object {
            "backgroundColor": "#FFFFFF",
            "marginTop": 16,
          },
        ]
      }
    >
      <View
        accessibilityLabel="GotoSignInFromBenefits"
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": undefined,
            "borderRadius": 4,
            "height": "100%",
            "justifyContent": "center",
            "opacity": 1,
            "width": 718,
          }
        }
        testID="GotoSignInFromBenefits"
      >
        <Text
          numberOfLines={1}
          style={
            Array [
              Object {
                "fontFamily": "SofiaProRegular",
                "fontSize": 17,
                "textAlign": "center",
              },
              Object {
                "color": "#2B342C",
              },
            ]
          }
        >
          Sign In
        </Text>
      </View>
    </View>
  </View>
</View>
`;
